packages/http-client-csharp/generator/Microsoft.Generator.CSharp.Input/src/InputTypes/Serialization/TypeSpecInputModelTypeConverter.cs

```diff
@@ -105,10 +105,14 @@ public static InputModelType CreateModelType(ref Utf8JsonReader reader, string?
             {
                 model.Properties = properties;
             }
-            if (discriminatedSubtypes != null)
+            if (discriminatedSubtypes?.Count > 0)
```

Comment:
Why did we need to change this check?

Reply:
ah, this is remnants of another approach I initially took. I have reverted this.